schema_based_on_diagram.sql

```diff
@@ -0,0 +1,71 @@
+create table patients (
+id SERIAL PRIMARY KEY,
+name VARCHAR(100),
+date_of_birth DATE
+);
+
+create table medical_histories (
+id SERIAL PRIMARY KEY,
+admitted_at TIMESTAMP,
+patient_id INT,
+status VARCHAR(100),
+CONSTRAINT FK_patient FOREIGN KEY (patient_id) REFERENCES patients(id)
+);
+
+create table treatments (
+id SERIAL PRIMARY KEY,
+type VARCHAR(100),
+name VARCHAR(100)
+);
+
+create table invoices (
+id SERIAL PRIMARY KEY,
+total_amount DECIMAL,
+generated_at TIMESTAMP,
+payed_at TIMESTAMP,
+CONSTRAINT FK_medical_history FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+create table invoices_items (
+id SERIAL PRIMARY KEY,
+unit_price DECIMAL,
+quantity INT,
+total_price DECIMAL,
+invoice_id INT,
+treatment_id INT,
+CONSTRAINT FK_invoice FOREIGN KEY (invoice_id) REFERENCES invoices (id),
+CONSTRAINT FK_treatment FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+);
+
+
+CREATE TABLE invoice_treatment (
+  invoice_id INT,
+  treatment_id INT,
+  PRIMARY KEY (invoice_id, treatment_id),
+  CONSTRAINT FK_invoice FOREIGN KEY (invoice_id) REFERENCES invoices (id),
+  CONSTRAINT FK_treatment FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+);
```

Comment:
- [ ] This table doesn't exist in the ERD please 🙏🏽 remove it from your schema because it isn't a join table, kindly pay attention in the unions there are one to many relationship between invoices and invoce_items and treatments and invoice_items no many to many as there is between medical_histories and treatments. The invoice_items table is enough and correct.